src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/ThreadScheduler.kt

```diff
@@ -67,6 +67,10 @@ sealed class BlockingReason {
     data class  ThreadJoin(val joinedThreadId: ThreadId) : BlockingReason()
 }
 
+fun BlockingReason.allowsSpuriousUnblocking(): Boolean =
+    this is BlockingReason.Parked       ||
```

Comment:
Is that always the case?

Reply:
Got it. Then I have two separate questions:

1. Can we rename it to `allowsSpuriousWakeUps`
2. How can we leave a live lock? (see the line below: `this is BlockingReason.LiveLocked`)

Reply:
1. Yes, I'll rename. 
2. No, live lock does not block currently, you right. I'll remove it.

Reply:
Oh, no, regarding (2), currently the code is:

https://github.com/JetBrains/lincheck/blob/7f4d8d4a418c8b6c6d8039ee7b79afa49ab08778/src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/managed/ManagedStrategy.kt#L536

So in essence I just extended it to `BlockingReason.Parked`.

Reply:
I reverted it